aws_sagemaker_studio/frameworks/pytorch_horovod_mpi/pytorch_horovod_mpi.ipynb

```diff
@@ -0,0 +1,407 @@
+    "# in the container by default, we're temporarily disabling the loading bar.\n",
+    "def gen_bar_updater():\n",
+    "    return lambda count, block_size, total_size: None\n",
+    "torchvision.datasets.utils.gen_bar_updater = gen_bar_updater\n",
+    "\n",
+    "dataset = torchvision.datasets.MNIST(\n",
+    "    root='./data',\n",
+    "    transform=torchvision.transforms.ToTensor(),\n",
+    "    download=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Previewing the Data\n",
+    "\n",
+    "Let's preview what the input images look like. We can do this using `matplotlib`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "dataloader = torch.utils.data.DataLoader(dataset, batch_size=4)\n",
+    "images, labels = next(iter(dataloader))\n",
+    "grid = torchvision.utils.make_grid(images)\n",
+    "plt.imshow(grid.permute(1, 2, 0))  \n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Uploading the Data\n",
+    "\n",
+    "We use the `sagemaker.Session.upload` function to upload our datasets to an S3 location. The return value, `inputs`, identifies the location of our data.\n",
+    "\n",
+    "To upload our data to S3, we need to specify an S3 bucket and prefix. These should be within the same region as the Notebook Instance, training, and hosting."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "from sagemaker.s3 import S3Uploader\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "bucket = sagemaker_session.default_bucket()\n",
+    "prefix = 'pytorch-horovod-mpi-example'\n",
+    "inputs = S3Uploader.upload('data', 's3://{}/{}/data'.format(bucket, prefix))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Entrypoint Script\n",
+    "\n",
+    "In order to use Horovod with OpenMPI, we use a shell script as the training entry point. The shell script invokes our Python training script as part of an Open MPI command. You can learn more about running Horovod with MPI at https://horovod.readthedocs.io/en/stable/mpirun.html.\n",
+    "\n",
+    "Here's what the shell script looks like:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!cat src/train.sh"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We're using 4 GPUs because that's the number of GPUs on a ml.p2.8xlarge instance, but if you're using a different instance type you may need to change the value of the`-np` parameter."
```

Comment:
```suggestion
    "We're using 4 GPUs because that's the number of GPUs on a ml.p2.8xlarge instance, but if you're using a different instance type you may need to change the value of the `-np` parameter."
```